packages/expo-clipboard/android/src/main/java/expo/modules/clipboard/ClipboardModule.kt

```diff
@@ -3,39 +3,77 @@ package expo.modules.clipboard
 import android.content.Context
 import android.content.ClipData
 import android.content.ClipboardManager
+import android.os.Bundle
+import android.util.Log
+import expo.modules.core.utilities.ifNull
 
-import expo.modules.core.ExportedModule
-import expo.modules.core.ModuleRegistry
-import expo.modules.core.Promise
-import expo.modules.core.interfaces.ExpoMethod
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
 
-class ClipboardModule(context: Context) : ExportedModule(context) {
-  private val NAME = "ExpoClipboard"
+private const val moduleName = "ExpoClipboard"
+private const val clipboardChangedEventName = "onClipboardChanged"
+private val TAG = ClipboardModule::class.java.simpleName
 
-  override fun getName(): String {
-    return NAME
-  }
+class ClipboardModule : Module() {
+  override fun definition() = ModuleDefinition {
+    name(moduleName)
 
-  override fun onCreate(moduleRegistry: ModuleRegistry) {
-    ClipboardEventEmitter(context, moduleRegistry)
-  }
+    function("getStringAsync") {
+      val clip = clipboardManager?.primaryClip?.takeIf { it.itemCount >= 1 }
+      clip?.getItemAt(0)?.text ?: ""
+    }
+
+    function("setString") { content: String ->
+      val clip = ClipData.newPlainText(null, content)
+      clipboardManager?.setPrimaryClip(clip)
+    }
+
+    events(clipboardChangedEventName)
+
+    onCreate {
+      clipboardEventEmitter = ClipboardEventEmitter()
+    }
 
-  @ExpoMethod
-  fun getStringAsync(promise: Promise) {
-    val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
-    val clip = clipboard.primaryClip
-    if (clip != null && clip.itemCount >= 1) {
-      promise.resolve(clip.getItemAt(0).text)
-    } else {
-      promise.resolve("")
+    onActivityEntersBackground {
+      clipboardEventEmitter.pauseListening()
+    }
+
+    onActivityEntersForeground {
+      clipboardEventEmitter.resumeListening()
     }
   }
 
-  @ExpoMethod
-  fun setString(content: String, promise: Promise) {
-    val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
-    val clip = ClipData.newPlainText(null, content)
-    clipboard.setPrimaryClip(clip)
-    promise.resolve(null)
+  private val context
+    get() = requireNotNull(appContext.reactContext)
+
+  private val clipboardManager: ClipboardManager?
+    get() = context.getSystemService(Context.CLIPBOARD_SERVICE) as? ClipboardManager
```

Comment:
I think we should throw something if `ClipboardManager` isn't present. For instance, you can inherit from `CodedException`. By doing this, you don't need to do anything more - promises will be automatically rejected.